contracts/andromeda_gumball/src/contract.rs

```diff
@@ -0,0 +1,772 @@
+    // Add to list of NFTs
+    list.push(mint_msg.token_id.clone());
+
+    LIST.save(deps.storage, &list)?;
+    let mission_contract = contract.get_mission_contract(deps.storage)?;
+
+    let contract_addr = token_contract.get_address(deps.api, &deps.querier, mission_contract)?;
+
+    Ok(Response::new()
+        .add_attribute("action", "mint")
+        .add_message(WasmMsg::Execute {
+            contract_addr,
+            msg: encode_binary(&Cw721ExecuteMsg::Mint(mint_msg))?,
+            funds: vec![],
+        }))
+}
+
+fn execute_buy(deps: DepsMut, _env: Env, info: MessageInfo) -> Result<Response, ContractError> {
+    let status = STATUS.load(deps.storage)?;
+    // check gumball's status
+    require(status, ContractError::Refilling {})?;
+    let mut list = LIST.load(deps.storage)?;
+    let n_of_nfts = list.len();
+    // check if we still have any NFTs left
+    require(n_of_nfts > 0, ContractError::OutOfNFTs {})?;
+    // check if more than one type of coin was sent
+    require(
+        info.funds.len() == 1,
+        ContractError::InvalidFunds {
+            msg: "Only one type of coin is required (uusd).".to_string(),
+        },
+    )?;
+    let sent_funds = &info.funds[0];
+    // check for correct denomination
+    require(
+        sent_funds.denom == DENOM,
+        ContractError::InvalidFunds {
+            msg: "Only uusd is accepted".to_string(),
+        },
+    )?;
+
+    let state = STATE.load(deps.storage)?;
+    // check for correct amount of funds
+    require(
+        sent_funds.amount == state.price.amount,
+        ContractError::InsufficientFunds {},
+    )?;
+    let contract = CW721_CONTRACT.load(deps.storage)?;
+    let randomness_source = RANDOMNESS_PROVIDER.load(deps.storage)?;
+    let random_response: LatestRandomResponse =
+        deps.querier.query(&QueryRequest::Wasm(WasmQuery::Smart {
+            contract_addr: randomness_source,
+            msg: encode_binary(&terrand::msg::QueryMsg::LatestDrand {})?,
+        }))?;
+    let randomness = Binary::to_base64(&random_response.randomness);
+    let vec = randomness.into_bytes();
+    let ran_vec: Vec<u64> = vec.iter().map(|x| *x as u64).collect();
+    // Concatinating the elements of the random number would yield an unworkably large number
+    // So I opted for the sum, which is still random and large enough to work with modulus of list's length
+    let mut random_number: u64 = ran_vec.iter().sum();
+    // In case the random number is smaller than the number of NFTs
+    while random_number < n_of_nfts as u64 {
+        random_number *= 2;
+    }
+    // Use modulus to get a random index of the NFTs list
+    let index = random_number as usize % n_of_nfts;
+    // Select NFT & remove it from list at the same time. Used swap_remove since it's more efficient and the ordering doesn't matter
+    let random_nft = list.swap_remove(index);
+    LIST.save(deps.storage, &list)?;
+    let contract_address = contract.identifier;
```

Comment:
This wasn't quite what I meant. What I meant was more along the lines of 
```
let mission_contract = ADOContract::default().get_mission_contract(deps.storage)?;
let contract_addr = contract.get_address(deps.api, &deps.querier, mission_contract)?;
```
What this does is query the mission contract to get the mapping of the address if it exists there. Also probably best to rename `contract` here to `token_contract` to not be ambiguous with `ADOContract` which often has that name.